exporter/metric/metric.go

```diff
@@ -606,20 +611,20 @@ func toNonemptyTimeIntervalpb(start, end time.Time) (*monitoringpb.TimeInterval,
 	}, nil
 }
 
-func histToTypedValue(hist metricdata.HistogramDataPoint) *monitoringpb.TypedValue {
+func histToTypedValue[N int64 | float64](hist metricdata.HistogramDataPoint[N]) *monitoringpb.TypedValue {
 	counts := make([]int64, len(hist.BucketCounts))
 	for i, v := range hist.BucketCounts {
 		counts[i] = int64(v)
 	}
-	var mean float64
-	if !math.IsNaN(hist.Sum) && hist.Count > 0 { // Avoid divide-by-zero
-		mean = hist.Sum / float64(hist.Count)
+	var mean N
+	if !math.IsNaN(float64(hist.Sum)) && hist.Count > 0 { // Avoid divide-by-zero
+		mean = hist.Sum / N(hist.Count)
```

Comment:
maybe we should keep these as forcing `float64`. if this code is modified to divide by an integer it will change the output